### PR TITLE
Fix not product not appearing in search results after having enabled it

### LINKED ISSUE #9663: [BOOM-5380] Can not search a disabled product after enabling it from product list
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-5380).

- _**Reporter:**_ khouloud.belguith
- _**Created at:**_ Thu, 19 Apr 2018 12:56:14 +0200

<p>Hi,</p>

<p>When we create a product and we save it without enabling it, after that, we change the status from the product list, this product is not indexed.<br/>
In fact, after enabling the product, in FO there is no product in the search result.</p>

<p>Best regards, Khouloud</p>
        
* How to reproduce the issue ?

1-Create product
<br/>
2-Save the product without enabling it
<br/>
3-Change the status of this product from the product list
<br/>
4-Go to the front office and search for the created product 
<br/>
5-The product does not exist 
<br/>
6-Go to BO =&gt; Search page, the product is missing in the indexing.

### PR DESCRIPTION
**Pull request by PeeyushAgrawal**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR fixes the issue #9663 
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #9663 
| How to test?  | Only a single change is required on the code end. By adding these changes, perform the below steps.
1. Create/save a new product in the offline stage from back-office catalog.
2. Now activate this product from the catalog product list.
3. Now the product will activate and now it will search the product.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15167)

### CODE REVIEW

File: themes/classic/templates/customer/_partials/block-address.tpl [hunk 1]
```diff
@@ -27,6 +27,8 @@
     <div class="address-body">
       <h4>{$address.alias}</h4>
       <address>{$address.formatted nofilter}</address>
+      {* To display the fields value extra added in a address from by using hook 'additionalCustomerAddressFields' *}
+      {hook h='displayAdditionalCustomerAddressFields' address=$address}
```
Comment (matthieu-rolland) [inline themes/classic/templates/customer/_partials/block-address.tpl]: Is this modification directly linked to this issue? #9663

**Review by matthieu-rolland — COMMENTED** [review]
Thank you for your contribution @PeeyushAgrawalWebkul  !

File: src/Adapter/Product/AdminProductDataUpdater.php [hunk 2]
```diff
@@ -80,6 +80,9 @@ public function activateProductIdList(array $productListId, $activate = true)
             }
             $product->active = ($activate ? 1 : 0);
             $product->update();
+            if (in_array($product->visibility, array('both', 'search')) && Configuration::get('PS_SEARCH_INDEXATION')) {
+                Search::indexation(false, $product->id);
+            }
```
Comment (PeeyushAgrawal) [inline src/Adapter/Product/AdminProductDataUpdater.php:87]: @matthieu-rolland 
Only this modification is directly linked to this PR.

File: themes/classic/templates/customer/_partials/block-address.tpl (hunk 1 above)
Comment (PeeyushAgrawal) [reply themes/classic/templates/customer/_partials/block-address.tpl]: @matthieu-rolland 
No, this modification is not linked to this issue.

Comment (matthieu-rolland) [reply themes/classic/templates/customer/_partials/block-address.tpl]: @PeeyushAgrawalWebkul Could you please remove this? You can put it in another dedicated pull request if you think this feature is important.

File: src/Adapter/Product/AdminProductDataUpdater.php (hunk 2 above)
Comment (matthieu-rolland) [reply src/Adapter/Product/AdminProductDataUpdater.php:87]: Ok, this looks fine, I will approve your pull request once you have removed the hook in the classic template 👍

Comment (matthieu-rolland) [reply src/Adapter/Product/AdminProductDataUpdater.php:87]: Hello @PeeyushAgrawalWebkul  !
Do you think you can remove the hook so I can approve your PR? If you can't I can take if from there if you want.

Comment (PeeyushAgrawal) [reply src/Adapter/Product/AdminProductDataUpdater.php:87]: Hello @matthieu-rolland 

Thanks,
Yes, Please proceed with the branch 
PeeyushAgrawalWebkul:solution-#9663

Comment (PeeyushAgrawal) [reply src/Adapter/Product/AdminProductDataUpdater.php:87]: Hello @matthieu-rolland 

I have removed the hook. Please check and proceed to merge the PR.

**Comment by matthieu-rolland** [comment]
@PeeyushAgrawalWebkul 

We're almost there, we just need you to do a rebase :)

**Comment by PeeyushAgrawal** [comment]
@matthieu-rolland 

Ok, thanks. Please proceed.

**Review by matthieu-rolland — APPROVED** [review]
lgtm

**Pull request merged**